README.md

```diff
@@ -75,15 +90,67 @@ From the extension preferences you will be able to set:
   - Preferred new password method ("generate" or "insert")
 
+##### Host application preferences
+If you use a customized `pass` installation: environment variables, customized repository path or extensions, you may have to [configure the host application accordingly](https://github.com/passff/passff-host#preferences).
+
+### Usage
+Once installed, you should have a new icon in your toolbar. Click the icon to browse your password repository or search using a **fuzzy matching** algorithm.
+
+##### Keyboard shortcuts
+The default shortcut to open the menu is <kbd>ctrl</kbd>+<kbd>y</kbd>.
+
+With the menu open, you can press <kbd>enter</kbd> to execute one of the following commands, according to your preferences:
+- Goto, fill and submit
+- Goto and fill
+- Fill and submit
+- Fill
+
+##### Input menu
+PassFF can *mark fillable input fields with the PassFF icon*. It adds an icon in the fields that PassFF can automatically fill. The icon is clickable and pops up a menu to select the password.
+PassFF fills the input fields and optionally submit depending on your preferences. You can always override this behavior by clicking the pencil (Fill) or the paper plane (Fill & Submit).
+
+This feature can be disabled in the preferences.
+
+#### Contextual menu
+In *any* input field, fillable or not, you can access a contextual menu (right-click) in order to:
+  - Add the input field's name in the *Login input names* for (auto)filling,
+  - Select a password to fill the input fields.
+
+#### Adding new passwords
+In order to add a password in your repository, select the 'plus' (+) icon in the toolbar menu.
+
 ### Issues
-If you're having problems, the most common causes are misconfigured preferences or an improperly installation of the host application. You can get more information by [debugging the extension](docs/CONTRIBUTING.md).
+If you're having problems, the most common causes are misconfigured preferences or an incorrect installation of the host application. You can get more information by [debugging the extension](docs/CONTRIBUTING.md).
+
+First, [make sure the host application is installed correctly](https://github.com/passff/passff-host).
+
+Configure the script's execution parameters appropriately in the host app `passff.py`: E.g., set `COMMAND` to the path to the `pass` binary (if installed with homebrew, the default location is `/usr/local/bin/pass`). With those settings in place, the extension should be able to find your passwords.
+
+### Troubleshooting
+
+#### I use an old version of Firefox and I have weird behaviours
+PassFF is developed for the last version of **Firefox** (version 59 as of April 2018).
+PassFF should also work on previous versions above Firefox 48, which introduced stable support for [WebExtensions](https://blog.mozilla.org/addons/2016/04/29/webextensions-in-firefox-48/).
+However, with Firefox's API transition, PassFF might behave strangely. See https://github.com/passff/passff/issues/272 for an issue now resolved.
```

Comment:
You want to mention #272 as an example where WebExtensions API transition broke PassFF in the past, right? However, I cannot really think of an example where a user can learn something from reading that outdated issue description.